plugin.py

```diff
@@ -43,6 +43,20 @@ def on_settings_read(cls, settings: sublime.Settings) -> bool:
 
         return False
 
+    def on_ready(self, api) -> None:
+        api.on_notification("pyright/beginProgress", self.handle_begin_progress)
+        api.on_notification("pyright/endProgress", self.handle_end_progress)
+        api.on_notification("pyright/reportProgress", self.handle_report_progress)
+
+    def handle_begin_progress(self, params) -> None:
+        sublime.status_message("{}: Progress begins".format(self.package_name))
+
+    def handle_end_progress(self, params) -> None:
+        sublime.status_message("{}: Progress ends".format(self.package_name))
+
+    def handle_report_progress(self, params: List[str]) -> None:
+        sublime.status_message("{}: {}".format(self.package_name, "; ".join(params)))
```

Comment:
Strangely API documentation doesn't list `sublime.status_message`. But I guess it's a shortcut to `active_window().status_message` so it's fine (I guess ;)).

Reply:
Oops! I didn't even notice this and used it for quite a while...